### PR TITLE
Data from DB for puzzles, exception handling

### PR DESCRIPTION
**Pull request by AiniusG**

### CODE REVIEW

File: backend/backend/MathGame/Services/MathGameService.cs [hunk 1]
```diff
@@ -1,102 +1,141 @@
...
+    //             new Puzzle { Id = Guid.NewGuid(), Content = "10 - 7" },
+    //             new Puzzle { Id = Guid.NewGuid(), Content = "6 * 2" },
+    //             new Puzzle { Id = Guid.NewGuid(), Content = "8 / 2" }
+    //         };
+
+    //         _context.Puzzles.AddRange(seedPuzzles);
+    //         _context.SaveChanges();
+    //         Console.WriteLine($"Seeded {seedPuzzles.Count} puzzles into the database.");
+    //     }
+    // }
```
Comment (Aidas21J) [inline backend/backend/MathGame/Services/MathGameService.cs]: Dead code'as

File: backend/backend/MathGame/Services/MathGameService.cs [hunk 2]
```diff
@@ -1,102 +1,141 @@
...
+            Console.WriteLine("Failed to write to the log file.");
         }
     }
 
-    private void LoadPuzzlesFromFile()
+    public string? GetNextPuzzle()
     {
-        try
+        var puzzles = _cache.Get<List<string>>("Puzzles");
+        if (puzzles == null || puzzles.Count == 0) { return null; }
```
Comment (Aidas21J) [inline backend/backend/MathGame/Services/MathGameService.cs]: Personally, `return` rasyciau is naujos eil.

File: backend/backend/MathGame/Services/MathGameService.cs [hunk 3]
```diff
@@ -1,102 +1,141 @@
...
+        if (tokens.Length != 3) { return false; }
 
-            if (_puzzles.Count == 0)
-            {
-                Console.WriteLine("The puzzle file is empty.");
-            }
+        // Initialize operands with default values
+        int rightOperand = 0;
+        int leftOperand;
+        bool isValidOperands = int.TryParse(tokens[0], NumberStyles.Integer, CultureInfo.InvariantCulture, out leftOperand) &&
```
Comment (Aidas21J) [inline backend/backend/MathGame/Services/MathGameService.cs]: Probably geriau `isValidOperands` sukurt is dvieju kitu `bool`, nes dabar there's not that much difference ar ta kilometrine salyga `if`'e, ar ne.

File: backend/backend/Program.cs [hunk 4]
```diff
@@ -32,6 +32,7 @@ public static void Main(string[] args)
         // Add custom services for DI
 
         builder.Services.AddScoped<IAimTrainerGameService, AimTrainerGameService>();
+        builder.Services.AddMemoryCache();
```
Comment (Aidas21J) [inline backend/backend/Program.cs]: Geriau sita perkelt toliau nuo visu `AddScoped`.

**Review by Aidas21J — APPROVED** [review]
Keliose vietose had couple of code style nitpicks, bet other than that looks good.

File: backend/backend/MathGame/Services/MathGameService.cs (hunk 2 above)
Comment (AiniusG) [reply backend/backend/MathGame/Services/MathGameService.cs]: Pagal ideja dotnet turejo suformatuot, fixed :D

Comment (Aidas21J) [reply backend/backend/MathGame/Services/MathGameService.cs]: Yeah, bet jei ne akivaizdi formatavimo klaida, tai gali palaikyt kaip weird formatavimo stiliu ir netaisyt.

**Review by Dax0s — APPROVED** [review]
LGTM

**Pull request merged**